### ISSUE TITLE
funky pod name causes problem for dashboard

### ISSUE THREAD

**Issue by kgignatyev-inspur** [kubernetes/dashboard#3032]
##### Steps to reproduce

When clicking on pod pod-checkpointer-j2jxm-ip-172-31-48-156.us-east-2.compute.internal I see 500 error, but other links are fine
So this one causes 500
https://api.stackpoint.io/cluster/13530/kubeui/api/v1/proxy/namespaces/kube-system/services/kubernetes-dashboard/#!/pod/kube-system/pod-checkpointer-j2jxm-ip-172-31-48-156.us-east-2.compute.internal?namespace=kube-system

This one works fine
https://api.stackpoint.io/cluster/13530/kubeui/api/v1/proxy/namespaces/kube-system/services/kubernetes-dashboard/#!/pod/kube-system/kube-proxy-4nkzm?namespace=_all

##### Environment
```
Installation method: 
Kubernetes version: 1.10.2
Dashboard version: v1.8.3
Commit: ec1d7de4456e6a397c7f931f0a2bfc74a6ca2e9c
```

##### Observed result
Dashboard reported Unknown Server Error (500):
```
Unknown reference kind Pod

```

**Comment by jeefy** [comment]
@kgignatyev-inspur could you provide the YAML output of your pod? (`kubectl get pod pod-checkpointer-j2jxm-ip-172-31-48-156.us-east-2.compute.internal -o yaml`)?

If OwnerReferences is present this could be related to #3030 

Thanks!

**Comment by EliZucker** [comment]
I think that this issue was recently fixed here: f585597581f82c984b6caaf8fde15b701ca54004

**Comment by floreks** [comment]
Closing as this was fixed.
